### PR TITLE
chore: remove feature flag for task executor

### PR DESCRIPTION
**Pull request by lyondhill**
We now default task execution to pull from the user service a permission
set.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: task/backend/executor/executor.go [hunk 1]
```diff
@@ -292,16 +292,9 @@ func (e *Executor) createPromise(ctx context.Context, run *influxdb.Run) (*promi
...
-	var perm influxdb.PermissionSet
-	if e.flagger != nil && feature.UseUserPermission().Enabled(ctx, e.flagger) {
-		perm, err = e.ps.FindPermissionForUser(ctx, t.OwnerID)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if perm == nil {
-		perm = t.Authorization.Permissions
```
Comment (GeorgeMac) [inline task/backend/executor/executor.go:304]: I can't remember the context on this. But do we still need this bit? Do we have some like legacy tasks with a Token on them somewhere?

Comment (GeorgeMac) [reply task/backend/executor/executor.go:304]: I guess not since we added the owner migration. So all Tasks should have an owner ID right?

**Review by GeorgeMac — APPROVED** [review]
Looks good. Given we're happy removing that persisted token permission set.

File: task/backend/executor/executor.go (hunk 1 above)
Comment (lyondhill) [reply task/backend/executor/executor.go:304]: yep exactly. This is the last effort to remove our dependence on the `t.Authorization` which is populated using `maxPermissions`  Once this goes out we should have severed our dependence on URM's in the task system.

**Pull request merged**